dynawo/sources/API/PAR/PARMacroParSetFactory.cpp

```diff
@@ -0,0 +1,35 @@
+//
+// Copyright (c) 2015-2021, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+// This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools
+// for power systems.
+//
+
+/**
+ * @file PARMacroParSetFactory.cpp
+ * @brief PARMacroParSetFactory : implementation file
+ *
+ */
+
+#include "PARMacroParSetFactory.h"
+
+#include "PARMacroParSet.h"
+
+using std::string;
+
+using boost::shared_ptr;
+
+namespace parameters {
+
+shared_ptr<MacroParSet>
+MacroParSetFactory::newMacroParSet(const string& id) {
+  return shared_ptr<MacroParSet>(new MacroParSet(id));
```

Comment:
factory are generally used to hide an Impl class from the external world. As it is not the case here I am not sure if this is useful.

Reply:
to be honest, I had the same thought, but then I have observed that was done like this also for macroConnectors, macroStaticReference, etc.. if you wish, we can get rid of this (also for dyd objects?)
@adriguir @gautierbureau your opinions on this?

Reply:
after short discussion, factories have been removed for this commit